wsmaster/codenvy-hosted-api-resource/src/main/java/com/codenvy/resource/api/ResourceAggregator.java

```diff
@@ -70,33 +68,34 @@ public ResourceAggregator(Set<ResourceType> resourcesTypes) {
     }
 
     /**
-     * Returns list which is result of deduction {@code resourceToDeduct} from {@code totalResources}.
+     * Returns list which is result of deduction {@code resourceToDeduct} from {@code sourceResources}.
      *
-     * @param totalResources
-     *         the total resources
+     * @param sourceResources
+     *         the source resources
      * @param resourcesToDeduct
-     *         the resources which should be deducted from {@code totalResources}
+     *         the resources which should be deducted from {@code sourceResources}
      * @throws NoEnoughResourcesException
-     *         when {@code totalResources} list doesn't contain enough resources
+     *         when {@code sourceResources} list doesn't contain enough resources
      * @throws IllegalStateException
-     *         when {@code totalResources} or {@code deduction} contain resource with not supported type
+     *         when {@code sourceResources} or {@code deduction} contain resource with not supported type
      */
-    public List<? extends Resource> deduct(List<? extends Resource> totalResources,
+    public List<? extends Resource> deduct(List<? extends Resource> sourceResources,
                                            List<? extends Resource> resourcesToDeduct) throws NoEnoughResourcesException {
-        checkSupporting(totalResources);
+        checkSupporting(sourceResources);
         checkSupporting(resourcesToDeduct);
 
-        final Map<String, Resource> result = totalResources.stream()
-                                                           .collect(Collectors.toMap(Resource::getType,
-                                                                                     Function.identity()));
+        final Map<String, Resource> result = sourceResources.stream()
```

Comment:
usually you deal with resources as with a map<type, resouce>, and this code is repeated throughout all the codebase.
Let's consider collection -> map conversion as grouping by type, so `ResourceAggregator` can do that + it can describe and document override strategies for the resources with repeated type.